Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/launcher/LauncherActivityTest.kt

```diff
@@ -0,0 +1,105 @@
+package de.rki.coronawarnapp.ui.launcher
+
+import android.content.Intent
+import android.net.Uri
+import androidx.test.core.app.launchActivity
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import dagger.Module
+import dagger.android.ContributesAndroidInjector
+import de.rki.coronawarnapp.main.CWASettings
+import de.rki.coronawarnapp.storage.LocalData
+import de.rki.coronawarnapp.ui.onboarding.OnboardingActivity
+import de.rki.coronawarnapp.update.UpdateChecker
+import de.rki.coronawarnapp.util.ui.SingleLiveEvent
+import io.mockk.MockKAnnotations
+import io.mockk.Runs
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.spyk
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import testhelpers.BaseUITest
+import testhelpers.TestDispatcherProvider
+
+@RunWith(AndroidJUnit4::class)
+class LauncherActivityTest : BaseUITest() {
+
+    @MockK lateinit var updateChecker: UpdateChecker
+    @MockK lateinit var cwaSettings: CWASettings
+    lateinit var viewModel: LauncherActivityViewModel
+
+    @Before
+    fun setup() {
+        MockKAnnotations.init(this)
+        mockkObject(LocalData)
+        coEvery { updateChecker.checkForUpdate() } returns UpdateChecker.Result(isUpdateNeeded = false)
+        every { LocalData.isOnboarded() } returns false
+        viewModel = launcherActivityViewModel()
+        setupMockViewModel(
+            object : LauncherActivityViewModel.Factory {
+                override fun create(): LauncherActivityViewModel = viewModel
+            }
+        )
+
+        every { viewModel.events } returns mockk<SingleLiveEvent<LauncherEvent>>().apply {
+            every { observe(any(), any()) } just Runs
+        }
+    }
+
+    @After
+    fun teardown() {
+        clearAllViewModels()
+    }
+
+    @Test
+    fun testDeepLinkLowercase() {
```

Comment:
What are you testing here? There is no assert? 🤔

Reply:
Testing that the activity is launching when an intent with `data=https://coronawarn.app/E1/SOME_PATH_GOES_HERE` is fired

as you can see for cases below it will throw a `RuntimeException`